### DISCUSSION TITLE
Allow multiple contributor roles and expand list of roles

### DISCUSSION THREAD

**Discussion by KellyStathis** [datacite/datacite-suggestions#32] [Metadata Schema Suggestions]
### What is the problem that your suggestion solves?

These issues came up when working with collaborators who have been putting together credit metadata for a large set of data objects to be imported into [Repository]; we intend to mint DOIs (via [Consortium Lead]) for each of these objects. [Repository] has created its own credit metadata schema based on our requirements, which we have mapped to the DataCite schema. However, we have come across the following issues:

1. people take on numerous roles in a project and to represent that reality properly, it should be possible to assign them numerous roles. Otherwise, how are contributors supposed to decide which DataCite role they give themselves? From reading the trello board and various associated docs, it seems as though this shortcoming is due to the DataCite schema being encoded as XML.

2. contributor roles based on job titles assume that (a) there are universal descriptions for the activities performed by people with these job titles that apply to every person with that role, and (b) people read the documentation, especially for obscure titles like ""WorkPackageLeader"", or terms with numerous potential definitions, such as ""ProjectLeader"" or ""ProjectManager"" (based on my own experience, people in these roles may be active participants in a project or may just be a name on the grant). A better way to represent contributions to a project would be to capture the activities that people perform, as the CRediT ontology does. This leaves less room for interpretation and potential error.

3. even though [Repository]'s credit metadata schema has an expanded list of contributor roles (including the terms from CRediT), our collaborators found they were still unable to accurately represent the ways in which some people contributed to a project. I see that there is a proposal to incorporate the CRediT terms in a future DataCite release, but CRediT has not been updated in quite a while, and there are still project activities that are not covered adequately by the two vocabularies."

### What solution might meet your needs?

"contributorRole":
1. allow multiple roles per person
2. change the current ""job title""-based roles to roles based on activities
3. expand the range of roles available"

### What alternatives have you tried or considered?

[Repository] has its own Credit Metadata Schema, which is heavily based on the DataCite schema but covers some areas where the DC schema falls short or does not represent reality appropriately (particularly for data citation). Obviously it would be preferable not to have a proliferation of citation standards à la https://xkcd.com/927/ !

### Is there anything else you would like to share?

*Suggestion was previously submitted by a user. For DataCite staff reference: https://docs.google.com/spreadsheets/d/1ItM7bYocla83TCYmHI9oZ9Kgt7X0Lznfv6Zls7Gmi88/edit?resourcekey=&gid=1473745305#gid=1473745305*

### What group(s) would benefit from your suggestion?

- [ ] DataCite members registering DOIs and metadata and administering accounts
- [ ] Developers building integrations with DataCite services
- [ ] Harvesters who index DataCite DOIs and metadata
- [ ] Individuals or institutions analyzing DataCite DOIs and metadata
- [ ] DataCite staff
- [ ] Other

**Comment by tedhabermann** [comment]
This is really (at least) two items:
1. Supporting multiple contributorTypes (+1)
2. Adding new contributorTypes (more details required)

**Reply by ialarmedalien** [reply]
Re: adding new contributor roles:

The existing [Contributor Role Taxonomy](https://credit.niso.org), or CRediT, captures some roles that DataCite's controlled vocab does not, and it is used by a number of groups similar to [Repository]. However, CRediT does not cover all areas of contribution, and the other main alternative, the [Contributor Role Ontology](https://github.com/data2health/contributor-role-ontology), is too granular for many users. There have been community suggestions for additional roles to add to CRediT, but it's unclear who "owns" CRediT and what the procedure for updating it would be.

**Reply by tedhabermann** [reply]
The Metadata Working Group had many long discussions about the CRediT ontology and other related ontologies. If I remember it correctly, we concluded that CRediT was really focused on journal articles, i.e. generally too specific for DataCite. I like the Contributor Role Ontology myself... but I agree it is very granular (as is the work in this case)... 

I created a crosswalk that got contributed to the discussion and is attached here. As you can see, this was written over 5 years ago so it is certainly out of date. Certainly not definitive, but maybe a useful starting point. 
[Contributor Roles Crosswalk.pdf](https://github.com/user-attachments/files/20093630/Contributor.Roles.Crosswalk.pdf)

**Reply by rebeccaringuette** [reply]
The crosswalk looks useful. Thanks! 
But, what if DataCite would allow multiple roles _and_ roles to be indicated from other taxonomies both in the contributor and the creator fields. For a person that is a contributor, the requirement from DataCite could be that one of their roles be chosen, but then roles from other taxonomies could also be chosen. Additional DataCite roles could also be added. Similarly, a person in the creator container would be a 'creator' (as the required DataCite role) and be able to indicate additional roles from other taxonomies (e.g. some relevant item from CRediT for a paper).  Providing this flexibility should allow those people to provide proof of their contributions to their employers in performance review even if DataCite doesn't host the taxonomy the role is chosen from. Would this work?

**Reply by tedhabermann** [reply]
Rebecca - I am very familiar with the approach that you are describing as it is used in the ISO 191* standards. The DataCite Metadata Working Group has discussed it as a option and it clearly adds infinite flexibility. At the same time, it is a fundamental change in the approach to codelists which comes with considerable challenges for analysis and retrieval. Users of DataCite metadata no longer have a complete list of codelist values available without an expensive search and a potentially complicated multi-faceted crosswalk. At the same time, I think an ORCID profile might work better in a performance review then DataCite metadata. A more complicated approach to codelists in DataCite would also need to be considered in terms of its impact on ORCID.

In any case, the Metadata Working Group has critical input to this idea. They always have a variety of inputs...

**Reply by rebeccaringuette** [reply]
Understood. Would the idea of requiring each person to have a DataCite role simplify this? Then the infinite range of roles is at least tied to a known range managed by DataCite so that the search can be simpler.

**Reply by tedhabermann** [reply]
I agree, that is an interesting option. They all have + and-...

I like multiple contributor roles and that seems relatively doable...

BTW, an idea that I like is deprecating creator, funder, and publisher and making them all contributor roles to focus all people and organizations in the same section of the metadata. Of course that also if fraught...

**Reply by ialarmedalien** [reply]
Multiple contributor roles for each contributor seems like a no-brainer to me, and requires relatively little change to the schema and to downstream consumers (it's just a list instead of a single value).

I really like @tedhabermann's suggestion of deprecating funder, publisher, and creator and just using contributor + contributor role for everything. This is something that could easily be done via remapping. In particular, I find the contributor <=> creator / author to be unnecessary and the cause of strife as there are unspoken assumptions about the superiority of one role over the other, and who gets what is more down to office politics than work contributed to a project.

**Reply by rebeccaringuette** [reply]
If that route is pursued, then we would need to either determine what contributor roles would be authors listed in the citation (hard) OR have the option of the user indicating what people should be listed as authors vs contributors (easier) (e.g. based on the significance of the contribution). I imagine merging the Creator and Contributor containers in DataCite, requiring that at least one DataCite contributor type be included per person (which gives the opportunity for each person to be indicated as an author or not), and allowing for additional contributor roles to be indicated whether from DataCite or other taxonomies. Then, a person can be recognized both for multiple types of contributions _and_ be distinguished from others based on the significance of that contribution (e.g. author vs not).
I like the current distinguishing approach to separate out funders and publishers simply because it makes it easier to pull those out in an API call (not that I have tried this). Any new approach for those two concepts would need some similarly simple way to perform the same action.

**Comment by rebeccaringuette** [comment]
I just created a discussion on a related topic that suggests allowing users to indicate role using external taxonomies such as CRediT instead of updating the list, and adding the same feature to the creator container. [Discussion 169](https://github.com/datacite/datacite-suggestions/discussions/169)

**Comment by rebeccaringuette** [comment]
Here is an example implementation showing how we could represent role names from both CRediT and allcontributors.org. The implementation for contributors could be identical, just change 'creator' to 'contributor' everywhere it is used. I think the only difference between the two fields would be that people/orgs listed under 'creators' would be included in the formal generated citation while people under 'contributors' would not. If that difference in the citation generation is not desired, then we could just combine the two into one field (crazy idea, I know, but interesting). Also posted on Discussion 169. 
```
<creators>
<creator>
<creatorName nameType="Personal">Garcia, Sofia</creatorName>
<givenName>Sofia</givenName>
<familyName>Garcia</familyName>
<nameIdentifier schemeURI="https://orcid.org/" nameIdentifierScheme="ORCID">0000-0001-5727-2427</nameIdentifier>
<affiliation affiliationIdentifier="https://ror.org/03efmqc40" affiliationIdentifierScheme="ROR" schemeURI="https://ror.org">Arizona State University</affiliation>
<contributionType xml:lang="en" contributionTypeScheme="CRediT Contributor Roles" contributionTypeSchemeURI="https://credit.niso.org/contributor-roles/" contributorTypeValueURI="https://credit.niso.org/contributor-roles/funding-acquisition/">Funding acquisition</contributionType>
<contributionType xml:lang="en" contributionTypeScheme="All Contributors" contributionTypeSchemeURI="https://allcontributors.org/">fundingFinding</contributionType>
</creator>
<creator>
<creatorName xml:lang="en" nameType="Organizational">California Digital Library</creatorName>
<nameIdentifier schemeURI="https://ror.org/" nameIdentifierScheme="ROR">https://ror.org/03yrm5c26</nameIdentifier>
</creator>
</creators>
```

**Reply by KellyStathis** [reply]
Thanks for this example implementation!

We've received suggestions to merge Creators and Contributors in the past. Is maintaining a distinction important to you/your organization? (As in, people/orgs in the formal generated citation vs. people/orgs not in the formal generated citation.)

**Reply by rebeccaringuette** [reply]
Yes. For example, we currently include repository staff as contributors based on improvements to the metadata, but we don't consider them creators for the datasets. That credit solidly belongs to the missions and researchers that generated the datasets. Having the distinction between creators being included in the citation versus contributors is something we would like to keep. It would be much easier if they both had identical structures, though.

**Reply by rebeccaringuette** [reply]
but also see Ted's comment and the idea following.

**Reply by ialarmedalien** [reply]
@rebeccaringuette I'm wondering what the value of being listed as a "contributor" but not having one's name in the citation list for a resource is. How widespread is knowledge that there is an extra "contributors" field? If you were a "contributor" to a project and wanted to put the reference on your resume / LinkedIn / etc., anyone looking up the DOI would think you'd just made it up.

Is there a difference in the _type_ of contribution made by creators vs by contributors - by the role that they play on the project or the type of work they do?

**Reply by rebeccaringuette** [reply]
My understanding is that the importance of the contribution to the overall product is the distinguishing line between creators and contributors, which is more nuanced that what role the person had.  For data, a person at a data repository that contributed to the metadata describing the dataset, perhaps including documentation feedback, did contribute to the dataset by improving FAIR, but did not contribute in a way that changed the science communicated by the dataset, and so is correctly distinguished as a contributor but not a creator. For software, some might consider those that helped with small contributions to the documentation (e.g., typo fixes) as contributors but not creators. So, the exact place that distinguishing line falls for a given item will likely be different, even for items of the same type. On the other end of the spectrum, some consider all contributors as creators (e.g., [Sunpy](https://doi.org/10.5281/zenodo.17162925)), but this is not always the case.  

The DOI metadata will have both creators and contributors in it where a distinction is desired, and both should be presented on the respective landing page appropriately (e.g., the creators listed separate from and above the contributors). This way, someone that clicks the DOI for a given object will see both lists of people, but the citation is also appropriate.

**Reply by ialarmedalien** [reply]
> The DOI metadata will have both creators and contributors in it where a distinction is desired, and both should be presented on the respective landing page appropriately (e.g., the creators listed separate from and above the contributors). This way, someone that clicks the DOI for a given object will see both lists of people, but the citation is also appropriate.

This might be an ideal scenario, but it requires anyone hosting DOI'd content to update their implementation of the display of citations, both for a resource, and for any references that resource cites (e.g. the references for a dataset or a paper). That is going to be difficult to enforce unless there is significant motivation for the org hosting the resource and references, such as a funder mandating updates to citation display. Realistically, very few people are going to go to the DataCite API to download the full credit metadata for a resource; the majority will rely on what they see at the bottom of a paper or what is generated by their citation management tool.

**Reply by ialarmedalien** [reply]
> My understanding is that the importance of the contribution to the overall product is the distinguishing line between creators and contributors [...]

My concern with using this kind of subjective distinction is that it is very easily influenced by bias (see the existing citation gap between men and women, and the greater gap for women of colour), office politics, and other factors unconnected to participation in generating the resource.

> On the other end of the spectrum, some consider all contributors as creators (e.g., [Sunpy](https://doi.org/10.5281/zenodo.17162925)), but this is not always the case.

If some groups do make a distinction and others don't, having two separate terms becomes worthless without additional information on how the distinction is defined and enforced.

**Reply by rebeccaringuette** [reply]
That is up to the creators to provide.

**Comment by tedhabermann** [comment]
And, of course, creators have contributorType = “creator” (which doesn’t exist currently), so they can be identified for the citation…

Ted

**Reply by rebeccaringuette** [reply]
That could be an elegant way to combine the two fields into one. To ensure metadata creators distinguish who should be part of the citation and who should not, DataCite could require either 'Creator' or 'Contributor' as one of the contributorRole entries for each person. There would have to be support for multiple contributor roles per person so that we could indicate the role(s) in addition to this distinction. For example,
```
<contributors>
<contributor>
<contributorName nameType="Personal">Garcia, Sofia</contributorName>
<givenName>Sofia</givenName>
<familyName>Garcia</familyName>
<nameIdentifier schemeURI="https://orcid.org/" nameIdentifierScheme="ORCID">0000-0001-5727-2427</nameIdentifier>
<affiliation affiliationIdentifier="https://ror.org/03efmqc40" affiliationIdentifierScheme="ROR" schemeURI="https://ror.org">Arizona State University</affiliation>
<contributionType xml:lang="en" contributionTypeScheme="CRediT Contributor Roles" contributionTypeSchemeURI="https://credit.niso.org/contributor-roles/" contributorTypeValueURI="https://credit.niso.org/contributor-roles/funding-acquisition/">Funding acquisition</contributionType>
<contributionType xml:lang="en" contributionTypeScheme="All Contributors" contributionTypeSchemeURI="https://allcontributors.org/">fundingFinding</contributionType>
<contributionType xml:lang="en" contributionTypeScheme="DataCite">Creator</contributionType>
</contributor>
<contributor>
<contributorName xml:lang="en" nameType="Organizational">California Digital Library</creatorName>
<nameIdentifier schemeURI="https://ror.org/" nameIdentifierScheme="ROR">https://ror.org/03yrm5c26</nameIdentifier>
<contributionType xml:lang="en" contributionTypeScheme="DataCite">Contributor</contributionType>
</contributor>
</contributors>
```

**Comment by tedhabermann** [comment]
FYI, there are a lot of different kinds of contributors (https://datacite-metadata-schema.readthedocs.io/en/4.6/appendices/appendix-1/contributorType/) and they show up with ORCIDs in Commons…

The types are also much more useful for datasets than the CRediT ontology IMHO…

Ted

**Reply by rebeccaringuette** [reply]
Yes, agreed. Nothing is perfect, but the list is pretty good for datasets. There does still need to be support for other contributor role names since DataCite DOIs are used for other types of things, especially software.

**Comment by rebeccaringuette** [comment]
While we wait for support for this to come in a new major version, what about adding a text field to support this, similar to what is done for the ResourceType field? e.g., 
```
<contributors>
    <contributor contributorType="DataCollector">"Data Curation"
        <contributorName nameType="Personal">Garcia, Sofia</contributorName>
        <givenName>Sofia</givenName>
        <familyName>Garcia</familyName>
        <nameIdentifier schemeURI="https://orcid.org/" nameIdentifierScheme="ORCID">0000-0001-5727-2427</nameIdentifier>
        <affiliation affiliationIdentifier="https://ror.org/03efmqc40" affiliationIdentifierScheme="ROR" schemeURI="https://ror.org">Arizona State University</affiliation>
    </contributor>
    <contributor contributorType="Other">"Project Administration"
        <contributorName xml:lang="en" nameType="Organizational">California Digital Library</contributorName>
        <nameIdentifier schemeURI="https://ror.org/" nameIdentifierScheme="ROR">https://ror.org/03yrm5c26</nameIdentifier>
    </contributor>
</contributors>
```
where the "Data Curation" term from [CRediT](https://credit.niso.org/) is added as free text and mapped to the "DataCollector" option in DataCite as a close enough match, and the "Project Administration" term from the same ontology is similarly added and mapped to the "Other" option in DataCite. Would that be a small enough change to be incorporated in a minor version? Also, does that provide enough of a 'band-aid' in the meantime?
